### PR TITLE
Indicate with a notification that the kernel is restarting

### PR DESCRIPTION
**Pull request by DavidKutu**
-disable restart command (and button) when the notebook is busy

For #1197

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

File: src/client/datascience/notebook/notebookEditor.ts [hunk 1]
```diff
@@ -233,6 +233,8 @@ export class NotebookEditor implements INotebookEditor {
         const kernel = this.kernelProvider.get(this.file);
 
         if (kernel && !this.restartingKernel) {
+            const promise = this.restartKernelInternal(kernel);
```
Comment (DonJayamanne) [inline src/client/datascience/notebook/notebookEditor.ts:236]: We have already started restarting the kernel here even before displaying the message
Also, if the user clicks `No` or cancels the prmopt, its too late, as we have restarted it.

**Review by DonJayamanne — CHANGES_REQUESTED** [review]

**Review by DonJayamanne — COMMENTED** [review]
Did we address the enablement of the button?

**Comment by DavidKutu** [comment]
> Did we address the enablement of the button?

No, it either doesn't work right, or it happens so fast that it doesn't make a difference.

File: src/client/datascience/jupyter/jupyterNotebook.ts [hunk 2]
```diff
@@ -168,6 +169,9 @@ export class JupyterNotebookBase implements INotebook {
     public get disposed() {
         return this._disposed;
     }
+    public get isRestarting() {
+        return this._isRestarting;
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/jupyterNotebook.ts:173]: I think we might want to discuss this issue as a team.
Not a fan of this. We have 2 way to determine the status of a kernel now.
This property and the status.

Comment (DonJayamanne) [reply src/client/datascience/jupyter/jupyterNotebook.ts:173]: I would prefer to have a single way of determining this, i.e. using the existing status.
E.g. when restarting a kernel, manually emit `Restarting` and supress all status emissions until it has restarted or restart has failed at which point we can send the current status of the kernel (which would most likely be in a busted state).

Comment (DonJayamanne) [reply src/client/datascience/jupyter/jupyterNotebook.ts:173]: @rchiodo @IanMatthewHuff what do you think?
Book a meeting to discuss how we address this?

**Review by DonJayamanne — CHANGES_REQUESTED** [review]
Blocking on the new `isRestarting` API.

File: src/client/datascience/jupyter/jupyterNotebook.ts (hunk 2 above)
Comment (IanMatthewHuff) [reply src/client/datascience/jupyter/jupyterNotebook.ts:173]: Catching up from my interview day. Is the crux of the issue here the way that we handle restart (by swapping to an existing session) doesn't line up with the actual status of the kernel?

Comment (DonJayamanne) [reply src/client/datascience/jupyter/jupyterNotebook.ts:173]: Yes, thus inn such cases the status isn't doing what we expect it to.

**Comment by DavidKutu** [comment]
Claudia's ruling: disabling is a nice to have. Just have the notification appear, and the process not happening a bunch of times if the user spams the button.

**Review by rchiodo — APPROVED** [review]

**Review by IanMatthewHuff — APPROVED** [review]

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**